aries_cloudagent/wallet/askar.py

```diff
@@ -348,12 +357,21 @@ async def get_local_did_for_verkey(self, verkey: str) -> DIDInfo:
 
         try:
             dids = await self._session.handle.fetch_all(
-                CATEGORY_DID, {"verkey": verkey}, limit=1
+                CATEGORY_DID, {"verkey": verkey}
             )
         except AskarError as err:
             raise WalletError("Error when fetching local DID for verkey") from err
         if dids:
-            return self._load_did_entry(dids[0])
+            ret_did = dids[0]
+            ret_did_info = ret_did.value_json
+            if len(dids) > 1 and ret_did_info["did"].startswith("did:peer:4"):
+                # if it is a peer:did:4 make sure we are using the short version
+                other_did = dids[1]  # assume only 2
+                other_did_info = other_did.value_json
+                if len(other_did_info["did"]) < len(ret_did_info["did"]):
+                    ret_did = other_did
+                    ret_did_info = ret_did.value_json
```

Comment:
Shouldn't this be `ret_did_info = other_did.value_json`?

Reply:
Could be either I guess, it's the same value

Reply:
Could we just keep it the same to alleviate confusion? I know it's a bit nitpicky but from an outside perspective I would assume there's a reason they're not referencing the same object.

Reply:
Sure I'll fix it

Reply:
Fixed :-P